pkgs/by-name/ex/exhibit/package.nix

```diff
@@ -9,24 +9,31 @@
   wrapGAppsHook4,
   desktop-file-utils,
   libadwaita,
+  itstool,
 }:
 
 python3Packages.buildPythonApplication rec {
   pname = "exhibit";
-  version = "1.2.0";
+  version = "1.4.2";
   pyproject = false; # Built with meson
 
   src = fetchFromGitHub {
     owner = "Nokse22";
     repo = "Exhibit";
-    rev = "v${version}";
-    hash = "sha256-yNS6q7XbWda2+so9QRS/c4uYaVPo7b4JCite5nzc3Eo=";
+    tag = "v${version}";
+    hash = "sha256-/dug7U8ei+gSdepILLqhnoIBhZ5QZePkREtCUl4p1Hs=";
   };
 
+  postPatch = ''
+    substituteInPlace src/logger_lib.py src/window.py \
+      --replace-warn 'os.environ["XDG_DATA_HOME"]' 'os.environ.get("XDG_DATA_HOME", os.path.expanduser("~/.local/share"))'
+  '';
+
```

Comment:
We should not disregard a user's XDG_DATA_HOME preference and hard code it to some path.

If the upstream code is this simply/dumb, please just patch it to do the right thing instead (create subdir under XDG_DATA_HOME and use that).

Reply:
I am not disregarding, I am adding a fallback.

If the variable is defined it will use the original value.

Reply:
Ah, I read it wrong.

I'd prefer a patch or at least replace-fail.

Reply:
Patch is too hardcoded and replace-fail would become a build failure if eventually upstream fixes that.

Reply:
Yes that's the point.

Reply:
And bot bumps fail silently

Reply:
Basically, r-ryantm only notifies you when everything went right.

Reply:
We want to know when the patch no longer applies and needs to be updated. Making the build fail is the only reliable way to achieve that.